pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/MultiStageEngineIntegrationTest.java

```diff
@@ -99,6 +99,14 @@ public void testGeneratedQueries()
     super.testGeneratedQueries(false, true);
   }
 
+  @Test
+  public void testQueryOptions()
+      throws Exception {
+    String pinotQuery = "SET multistageLeafLimit = 1; SELECT * FROM mytable;";
+    String h2Query = "SELECT * FROM mytable";
```

Comment:
curious how did this test passed? don't you need to do `SELECT * FROM mytable LIMIT 1` for H2 query?

Reply:
Current test frame work only check H2 query contains the request for pinot query

Reply:
Hmm, this test cannot prove the option is taking effect. Can you also verify that the pinot query only returns one record?

Reply:
great catch @Jackie-Jiang. i think we can directly assert 1 row instead of comparing against h2

Reply:
Added a test to compare # of rows. PTAL.